source/extensions/filters/common/rbac/matchers.cc

```diff
@@ -23,9 +23,15 @@ MatcherConstPtr Matcher::create(const envoy::config::rbac::v3::Permission& permi
     return std::make_unique<const OrMatcher>(permission.or_rules(), validation_visitor, context);
   case envoy::config::rbac::v3::Permission::RuleCase::kHeader:
     return std::make_unique<const HeaderMatcher>(permission.header(), context);
-  case envoy::config::rbac::v3::Permission::RuleCase::kDestinationIp:
-    return std::make_unique<const IPMatcher>(permission.destination_ip(),
-                                             IPMatcher::Type::DownstreamLocal);
+  case envoy::config::rbac::v3::Permission::RuleCase::kDestinationIp: {
+    auto matcher_result =
+        IPMatcher::create(permission.destination_ip(), IPMatcher::Type::DownstreamLocal);
+    if (!matcher_result.ok()) {
+      throwEnvoyExceptionOrPanic(
+          fmt::format("Failed to create IP matcher: {}", matcher_result.status().message()));
+    }
+    return std::move(matcher_result.value());
+  }
```

Comment:
Ideally this function would *also* be migrated to return a StatusOr, since the whole point of migrating to StatusOr is to get away from exceptions.

But if that cascades too far such that this is a good cutoff point for a step along that road, then I would recommend `THROW_IF_NOT_OK_REF` as a terser variation for the temporary interim-throw. (Here and in the other similar instances below.)

Reply:
Thanks for the suggestion! I'll replace the exception throwing with `THROW_IF_NOT_OK_REF` in all the IP matcher creations.